### DISCUSSION TITLE
Checking types on the server side

### DISCUSSION THREAD

**Discussion by fitoprincipe** [gee-community/gee_tools#271] [Ideas]
I was thinking in creating a Feature Request for [`ee.Dictionary.geetools.fromPairs`](https://github.com/gee-community/geetools/blob/5ac04c777f4593af92fc04f7c3c96cdd02affc7b/geetools/Dictionary/__init__.py#L19) to be able to do

``` python
newdict = ee.Dictionary.fromPairs([{"a": 1}, {"b": 2}])
```

I know that some server side methods [accept different arguments to do the same](https://developers.google.com/earth-engine/apidocs/ee-dictionary), but I am not sure if we should try to mimic that behavior or just create as many methods as options.

To do it server side we can use [ee.Algorithms.ObjectType](https://developers.google.com/earth-engine/apidocs/ee-algorithms-objecttype). It'll also require to use [ee.Algorithms.If](https://developers.google.com/earth-engine/apidocs/ee-algorithms-if)

What do you think @12rambau?

**Comment by 12rambau** [comment]
what would be the difference from the `fromLists` constructor ? 

because if I can do this with your method: 

```python
params = [{"a": 1}, {"b": 2}]
newDict = ee.Dictionary.fromPairs(params) 
```

I could do the same with vanilla GEE methods and 1 extra line
```python
keys, values = zip(*[("a", 1}, ("b", 2)])
newDict = ee.Dictionary.fromLists(keys, values)
```

**Reply by fitoprincipe** [reply]
that's ok if the keys and values are client side, but if they are server side, it gets a bit more complicated, but of course not impossible..

 ``` python
thelist = ee.List([{"a":1}, {"b":2}])
keys = thelist.map(function(it){return ee.Dictionary(it).keys().getString(0)})
values = thelist.map(function(it){return ee.Dictionary(it).values().getNumber(0)})
thedict = ee.Dictionary.fromLists(keys, values)
```
or if you want to reduce it to one line maybe a `.iterate` would work, but...

**Reply by 12rambau** [reply]
My problem in general is I never faced this use case, what process leads to a list of (key, pair) stored as dict ?
